File: BAEKJOON/java/20362_유니대전_퀴즈쇼.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    public static int n, answer = 0;
    public static String winner, win_answer;
    public static List<Quiz> list = new ArrayList<>();

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        winner = st.nextToken();

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());

            String participant = st.nextToken();
            String answer = st.nextToken();

            Quiz quiz = new Quiz(participant, answer);

            list.add(quiz);
        }
    }

    private static void solution() throws IOException {
        for (int i = list.size() - 1; i >= 0; i--) {
            if (list.get(i).participant.equals(winner)) {
                win_answer = list.get(i).answer;
                continue;
            }

            if (list.get(i).answer.equals(win_answer)) {
                answer++;
            }
        }

        bw.write(String.valueOf(answer));
    }
}

class Quiz {
    String participant;
    String answer;

    public Quiz(String participant, String answer) {
        this.participant = participant;
        this.answer = answer;
    }
}

File: BAEKJOON/cpp/20362_유니대전_퀴즈쇼.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int n, answer = 0;
string winner, win_answer;
vector<pair<string, string>> v;

void input() {
	cin >> n >> winner;

	string a, b;
	for (int i = 0; i < n; i++) {
		cin >> a >> b;

		v.push_back({ a, b });
	}
}

void solution() {
	for (int i = v.size() - 1; i >= 0; i--) {
		if (v[i].first == winner) {
			win_answer = v[i].second;
			continue;
		}

		if (v[i].second == win_answer) {
			answer++;
		}
	}

	cout << answer;
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}